### DISCUSSION TITLE
`First 40 files shown. See all results` for non default branch yields no results

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#102143] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

1. Visit https://github.com/check-spelling/cspell-dicts/tree/85cd51ee3883004ea5601e6909838e6562fb24f1
2. Enter `shell` into the `Go to file` feature
    <img width="795" alt="image" src="https://github.com/community/community/assets/2119212/70fb85ba-a6e5-4f9c-bee5-df41e612fc03">
3. Click [`See all results`](https://github.com/search?q=repo%3Acheck-spelling/cspell-dicts%20path%3Ashell&type=code)
    <img width="638" alt="image" src="https://github.com/community/community/assets/2119212/54540db0-eba8-4096-826d-d1ddca1702ea">

---

the reason is that the default branch is mostly empty (it holds a workflow to maintain the other branches).

But if see all results won't work on non default branches, then it should be more cautious about suggesting it...

**Comment by muizzhaque** [comment]
Yea it is indeed a bug

**Comment by look** [comment, marked as answer]
@jsoref this is an edge case, I will report it to the team. The issue is that the fuzzy file finder works directly with the Git tree listing, which is why it works on any branch, tag, or commit. The "see all results" link is a code search, and code search only indexes the default branch.

**Reply by jsoref** [reply]
Yeah, I get it. One approach is to not be so confident about suggesting `see all results` when not on the default branch. Another would be a sanity check to trigger a check to see if there are >40 results in code search on the default branch before showing that text.

**Reply by jsoref** [reply]
Fwiw, the fuzzy-file-finder behavior isn't what I wanted which is the most frustrating part about this. I was looking for a file or directory with the name `shell` in it, not something matching `/s.*h.*e.*l.*l/`.

Specifically, I wanted to find:
[https://github.com/check-spelling/cspell-dicts/blob/main/dictionaries/shell/dict/shell-all-words.txt](https://github.com/check-spelling/cspell-dicts/blob/aeb9da40d59de34c77422bd16e9900ca0534af6c/dictionaries/shell/dict/shell-all-words.txt)
